.github/workflows/docs.yml

```diff
@@ -14,14 +16,37 @@ on:
       - docs/**
       - CHANGELOG.rst
       - README.md
+  workflow_dispatch:
+    inputs:
+      tag:
+        required: true
 
 jobs:
-  docs:
+
+  build:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/setup-python@v4
         with:
           python-version: '3.11'
+      - run: echo ${{ github.event.push.tag || github.event.inputs.tag }}
       - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.push.tag || github.event.inputs.tag }}
       - run: python -m pip install .[docs]
       - run: python -m sphinx docs/ build/docs/
+      - uses: actions/upload-pages-artifact@v1
+        with:
+          path: build/docs/
+
+  deploy:
+    needs: build
+    permissions:
+      pages: write
+      id-token: write
+    environment:
+      name: github-pages
+    runs-on: ubuntu-latest
+    if: ${{ github.event.push.tag || github.event.inputs.tag }}
+    steps:
+      - uses: actions/deploy-pages@v1
```

Comment:
any reason not to use `v2` here? Latest

Reply:
No reason other than I copy-pasted most of this from a project of mines and I didn't check if there has been another version released since I wrote this the first time. Let me check if v2 has any important change before moving to it.

Reply:
Updated the actions to v2.